schemas/com.snowplowanalytics.snowplow/screen_view/jsonschema/1-0-1

```diff
@@ -0,0 +1,61 @@
+{
+	"$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+	"description": "Schema for a screen view event",
+	"self": {
+		"vendor": "com.snowplowanalytics.snowplow",
+		"name": "screen_view",
+		"format": "jsonschema",
+		"version": "1-0-1"
+	},
+	"type": "object",
+	"properties": {
+		"name": {
+			"type": "string",
+            "maxLength": 100,
+            "description": "The name of the screen viewed."
+		},
+		"type": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "The type of screen that was viewed e.g feed / carousel."
+        },
+        "screen_session_id": {
```

Comment:
Perhaps there's a better name for this that avoids confusing it with the session id?

Reply:
+1

Reply:
screen_instance_id maybe? This is still a flawed descriptor though would be keen to come up with something more accurate

Reply:
I am a little confused with this field in general:

Why would the screen ID be regenerated each session? I would assume there is a 1:1 mapping of screen IDs to screen name and screen name would persist past the session?

If this is an ID that is generated each session then we could just use the session_id out of the client_session_context?

Reply:
There are two things here - one is a screen ID which is like a page view id - new UUID for every screen view event.

This one is to account for apps which don't do a good job of naming their screens. It gives a screen ID to the screen which doesn't change (per session. There's no way to retain the ID after the app closes). So let's say you haven't assigned screen names, and the user journey is:

Screen 1 [set of events a] -> screen 2 -> screen 1 [set of events b]

This field allows you to aggregate across all events which occurred on screen 1 in the session, regardless of how many times the user left and came back to the screen within that session.

That's how I understand it anyway.

Reply:
So actually this should be screen_id but that's confusing too. 

screen_name_id? aux_screen_name maybe?

Reply:
> There's no way to retain the ID after the app closes

Ah, this field makes sense to me then!

Perhaps a name to imply that it is transient such as `screen_temp_id`

Reply:
I think `screen_name_id` makes most sense to me. It's an ID that's meant to be used instead of a name.

Reply:
Yeah my concern on that one is that it suggests that it's a 1:1 map to name, which it isn't. It's a 1:1 map to name + session_id.

`persisting_screen_id` is another option but shares this problem.

Reply:
I think we should be careful to show that it _doesn't_ persist. I'm trying to imagine a world where a user didn't do a good job of naming screens and relied on `screen_xxxx_id` thinking it will be a persisting ID. I think we need to make clear that it doesn't really replace screen name as it _wont_ persist.

Reply:
Agreed. The tricky part is balancing that with the fact that it does persist in comparison to the screen_id.

It persists for the session, it doesn't persist forever....

What about either `screen_in_session_id` or `in_session_screen_id`?

Reply:
`in_session_screen_id` gets my +1

Reply:
likewise +1

Reply:
The motivation here is true autotracking where an app designer doesn't even have to be careful around how their screens are named. So ideally they'd name everything well but I thought, if I'm dealing with all these view controllers, I can round them up when the app starts and assign an ID from the start. But the problem is that these won't be the same across users and sessions since they're randomly generated.

I've tried to figure if there's any problem with the way I'm generating and associating the (currently called) screen_session_id, and I think perhaps the best name is `screen_instance_id`, and perhaps more accurately `viewcontroller_instance_id`. I'm using associated objects in Obj-C, where you're able to attach an object to another object. So there no way to really associate an id in a definite way across all devices, basically the best I can do is an id that persists as long as an object exists.

It's conceivable that a class could get instantiated and then deallocated, so the `instance_id` would change there. Maybe another field for the class name of the viewcontroller would be appropriate?

I think these schemas should be general, but perhaps there should be platform specific names like `viewcontroller_instance_id`, and `viewcontroller_class_name`.

Reply:
I'm not really sure if this is better handled by documenting it really clearly so the behind-the-scenes stuff is clear or whether it's better to offer more schema fields. Wondering what y'all think of supplemental platform specific fields. IMO, they offer a clearer idea of what's actually being captured for auto-tracking:

- For Android:
  - `activity_id`
  - `activity_name`
  - `fragment_id`
  - `fragment_name`

- For iOS:
  - `viewcontroller_id`
  - `viewcontroller_class_name`
  - `viewcontroller_name`

And (as we already have in the schema draft) required fields that during auto-tracking arbitrarily take info from the above fields:
  - `screen_id`
  - `name`

Reply:
I think this likely confuses the issue for most consumers of the data. Think product manager rather than developer.

Reply:
ok agreed, it seems like tracking screenviews through activities is standard on android anyway